src/views/open-api-docs-view/components/property-details/index.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import classNames from 'classnames'
+import { MdxInlineCode } from 'components/dev-dot-content/mdx-components'
+import Badge from 'components/badge'
+import { IconCornerDownRight16 } from '@hashicorp/flight-icons/svg-react/corner-down-right-16'
+import s from './property-details.module.css'
+
+/**
+ * Types are shared by the "base" and "deeper" variations of
+ * the property details component.
+ */
+export type PropertyDetailsProps = {
+	name: string
+	type: string
+	isRequired?: boolean
+	description?: string // Plain text or HTML
+	nestedProperties?: PropertyDetailsProps[]
+	depth?: number
+}
+
+/**
+ * Render a top-level property, with optional nested properties.
+ *
+ * Future: will need to support "beta" badge for HCP Packer docs.
+ */
+export function PropertyDetails({
```

Comment:
This component is the bulk of the presentational work here. `.tsx` and `.css` files here are a lil longer than usual, but leaned toward keeping them co-located in one file since they're pretty interdependent, and have the whole potentially recursive rendering thing going on.